[PFX-577] Docs Update: Move CLI docs (#827)
godaddy/gasket: docs/configuration.md

sub-environments. For example, here's a theoretical `gasket.js`:

```js
export default makeGasket({
  someService: {
    enableSomeFeature: false
  },
  environments: {
    dev: {
      someService: {
        url: 'https://your-dev-service-endpoint.com'
      }
    },
    test: {
      someService: {
        url: 'https://your-test-service-endpoint.com'
      }
    },
    prod: {
      someService: {
        enableSomeFeature: true
      }
    },
    'prod.datacenter1': {
      someService: {
        url: 'https://your-datacenter1-prod-service-endpoint.com'
      }
    },
    'prod.datacenter2': {
      someService: {
        url: 'https://your-datacenter2-prod-service-endpoint.com'
      }
    }
  }
});
```

When accessing configuration in the `prod.datacenter1` environment, the base
configuration is augmented with the `prod` environment settings, then the
`prod.datacenter1` settings, resulting in:

```json
{
  "someService": {
    "enableSomeFeature": true,
    "url": "https://your-datacenter1-prod-service-endpoint.com"
  }
}
```

The last step is making sure that Gasket is told which environment it is running

> in when starting up. There are three ways of doing this. The first would be to
> set the `GASKET_ENV` environment variable. Alternatively, you can
> programmatically set an `env` property at the top level of your configuration,
> enabling you to use any arbitrary environment derivation logic.

Reviewer: ```suggestion
in when starting up. To do so, set the `GASKET_ENV` environment variable.
Alternatively, you can programmatically set an `env` property at the top level
of your configuration, enabling you to use any arbitrary environment derivation logic.
```

      }
    },
    test: {
      someService: {
        url: 'https://your-test-service-endpoint.com'
      }
    },
    prod: {
      someService: {
        enableSomeFeature: true
      }
    },
    'prod.datacenter1': {
      someService: {
        url: 'https://your-datacenter1-prod-service-endpoint.com'
      }
    },
    'prod.datacenter2': {
      someService: {
        url: 'https://your-datacenter2-prod-service-endpoint.com'
      }
    }
  }
});
```

When accessing configuration in the `prod.datacenter1` environment, the base
configuration is augmented with the `prod` environment settings, then the
`prod.datacenter1` settings, resulting in:

```json
{
  "someService": {
    "enableSomeFeature": true,
    "url": "https://your-datacenter1-prod-service-endpoint.com"
  }
}
```

The last step is making sure that Gasket is told which environment it is running
in when starting up. There are three ways of doing this. The first would be to
set the `GASKET_ENV` environment variable. The second would be to pass an `--env`
command line argument when running any `gasket` command. The third is to
programmatically set an `env` property at the top level of your configuration,

> which would enable you to use any arbitrary environment derivation logic.

Reviewer: The `--env` is no longer an option. We don't support programmatically setting `env`, though perhaps we could?

Looking at [this line](https://github.com/godaddy/gasket/blob/v7/packages/gasket-core/lib/index.js#L54), perhaps we could do something like:
```
const env = gasketConfig.env ?? getEnvironment();
```

Author: Should this be in another ticket or is it ok to add to this PR?

## Accessing Gasket configuration in your application

You can access the Gasket configuration in your application by importing the `gasket` object or through plugin hooks.

```js
// example-plugin.js
export const name = 'example-plugin';

export const hooks = {
  express: async function (gasket, express) {
    console.log(gasket.config)
  }
};

export default { name, hooks };

// app-code.js

> import gasket from './gasket';

Reviewer: Show the extension

```suggestion
import gasket from './gasket.js';
```

console.log(gasket.config);
```

## External configuration

### Authoring a config plugin

If a config plugin does not yet exist for your configuration platform, writing
one is straightforward. For Gasket-level configuration, hook the `configure`
event. Your hook will be passed a config object, and you'll return an augmented
config object or a Promise containing one for async config.

```js
// theoretical.js plugin
export default {
  name: 'theoretical',
  hooks: {
    async configure(gasket, baseConfig) {

>       const remoteConfig = await theoreticalClient.getConfig(baseConfig.theoretical.url);

Reviewer: The `configure` lifecycle is not long async - can you change up this example?
